code/modules/atmospherics/gasmixtures/gas_types.dm

```diff
@@ -40,32 +68,99 @@ GLOBAL_LIST_INIT(nonreactive_gases, typecacheof(list(/datum/gas/oxygen, /datum/g
 \*||||||||||||||||||||||||||||||||||||||||*/
 
 /datum/gas
+	/// A reference to the cached gas metadata list
+	var/list/meta_gas_list
+
+	/// A unique text id for this gas. Required to parse gas strings
 	var/id = ""
+	/// How much thermal energy is required to increase the temperature of one mole of this gas by 1 degree kelvin
 	var/specific_heat = 0
+	/// What this gas is referred to IC
 	var/name = ""
+	/// The icon state of the overlay used to indicate this gas's presence in the atmosphere
 	var/gas_overlay = "" //icon_state in icons/effects/atmospherics.dmi
+	/// How many moles are required before the gas overlay starts to show up. If null the gas is invisible
 	var/moles_visible = null
+	/// Whether or not the gas is dangerous. Used for logging
 	var/dangerous = FALSE //currently used by canisters
-	var/fusion_power = 0 //How much the gas accelerates a fusion reaction
-	var/rarity = 0 // relative rarity compared to other gases, used when setting up the reactions list.
+	/// How much the gas accelerates a fusion reaction. See /code/modules/atmospherics/machinery/components/fusion/hypertorus.dm for details
+	var/fusion_power = 0
+	/// Relative rarity compared to other gases, used when setting up the reactions list.
+	var/rarity = 0
+
+	// Condensation:
+	/// The maximum rate of condensation of this gas (as a function of total moles). If this is falsey the gas will not condense
+	var/cond_rate
+	/// The maximum temperature this gas will condense at
+	var/cond_temp_max
+	/// The temperature at and below which the gas will condense at it's maximum rate
+	var/cond_temp_min
+	/// The typepath of the reagent this gas condenses into. If this is null the reagents will not produce a gas when condensing
+	var/datum/reagent/cond_type
+	/// The amount of thermal energy released by one mole of this gas condensing
+	var/cond_heat = 0
+	/// A reference to the condensation event [CALLBACK] If this is truthy on init it will be repaced with the appropriate callback. Leave this as null
+	var/datum/callback/cond_event = null
```

Comment:
Why is this being done on gas_types? And why are you treating them as singletons rather then templates

Reply:
Because I want different gases to condense at different temperatures and rates. The singleton bit is just so it can generate the callback for condensation effects.